packages/netlify-cms-backend-git-gateway/src/implementation.ts

```diff
@@ -193,53 +255,52 @@ export default class GitGateway {
     return user && user.logout();
   }
   getToken() {
-    return this.tokenPromise();
+    return this.tokenPromise!();
   }
 
-  entriesByFolder(collection, extension) {
-    return this.backend.entriesByFolder(collection, extension);
-  }
-  entriesByFiles(collection) {
-    return this.backend.entriesByFiles(collection);
+  entriesByFolder(folder: string, extension: string, depth: number) {
+    return this.backend!.entriesByFolder(folder, extension, depth);
   }
-  fetchFiles(files) {
-    return this.backend.fetchFiles(files);
+  entriesByFiles(files: ImplementationFile[]) {
```

Comment:
Each backend had the same code to extract the files from the collection so I moved that to the core